### PR TITLE
OCPBUGS-20033: Make the OLMCatalogPlacement field immutable

### PR DESCRIPTION
**Pull request by jparrill**
**Which issue(s) this PR fixes**:
Fixes #[OCPBUGS-20033](https://issues.redhat.com/browse/OCPBUGS-20033)

### CODE REVIEW

File: api/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -434,6 +434,7 @@ type HostedClusterSpec struct {
 
 // OLMCatalogPlacement is an enum specifying the placement of OLM catalog components.
 // +kubebuilder:validation:Enum=management;guest
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
 type OLMCatalogPlacement string
```
Comment (muraee) [inline api/v1beta1/hostedcluster_types.go:438]:
```suggestion
// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
// +immutable
type OLMCatalogPlacement string
```
I think we need `// +immutable` for the docs.

**Comment by enxebre** [comment]
/hold
cc @relyt0925 @hasueki to validate you are ok enforcing this?

**Review by hasueki — APPROVED** [review]
/lgtm

>cc @relyt0925 @hasueki to validate you are ok enforcing this?

Checked with our teams and good from IBM Cloud perspective!

File: api/v1beta1/hostedcluster_types.go [hunk 2]
```diff
@@ -433,7 +433,10 @@ type HostedClusterSpec struct {
 }
 
 // OLMCatalogPlacement is an enum specifying the placement of OLM catalog components.
+//
 // +kubebuilder:validation:Enum=management;guest
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="OLMCatalogPlacement is immutable"
```
Comment (enxebre) [inline api/v1beta1/hostedcluster_types.go:438]: why is this here instead of in https://github.com/openshift/hypershift/blob/main/api/v1beta1/hostedcluster_types.go#L424-L427

**Comment by enxebre** [comment]
Thanks! Dropped a question https://github.com/openshift/hypershift/pull/3113#discussion_r1371585856. Also have we tested this?

**Comment by jparrill** [comment]
> Thanks! Dropped a question [#3113 (comment)](https://github.com/openshift/hypershift/pull/3113#discussion_r1371585856). Also have we tested this?

I though that the right place is the current one. It should be the other one? @enxebre

**Comment by jparrill** [comment]
PR Updated @enxebre. Let me know if there is any other change pending.
